veggieseasons/lib/screens/details.dart

```diff
@@ -238,15 +238,42 @@ class InfoView extends StatelessWidget {
 
 class DetailsScreen extends StatefulWidget {
   final int id;
+  final String restorationId;
 
-  DetailsScreen(this.id);
+  DetailsScreen({this.id, this.restorationId});
+
+  static String show(NavigatorState navigator, int veggieId) {
+    return navigator.restorablePush<void>(_routeBuilder, arguments: veggieId);
+  }
+
+  static Route<void> _routeBuilder(BuildContext context, Object arguments) {
```

Comment:
This bit is interesting to me, because it brings the type of route used to display the screen into the class itself, whereas that used to be something the launching screen was in control of (e.g. previously VeggieCard was responsible for deciding to show DetailsScreen as a dialog, and DetailsScreen itself was unconcerned with anything to do with how it was added to the navigation stack).

Is the pattern used here something we'd expect to see in most apps using State Restoration? It looks like the main reason for the change is to allow DetailsScreen to make sure its restoration ID to 'details', but I could be reading this wrong.

Reply:
A lot of people prefer to have the navigation info in the screens themselves, as you have it. I've seen plenty of apps that include route strings as constants in classes that define the Screens. I mostly wanted to make sure a) I understood what was going on, and b) whether this was a pattern you'd recommend for everyone.

No need to change it.